pkg/codegen/dotnet/gen.go

```diff
@@ -1158,10 +1158,23 @@ func (mod *modContext) genResource(w io.Writer, r *schema.Resource) error {
 		if fun.DeprecationMessage != "" {
 			fmt.Fprintf(w, "        [Obsolete(@\"%s\")]\n", strings.ReplaceAll(fun.DeprecationMessage, `"`, `""`))
 		}
-
 		fmt.Fprintf(w, "        public %s %s(%s)\n", returnType, methodName, argsParamDef)
-		fmt.Fprintf(w, "            => Pulumi.Deployment.Instance.Call%s(\"%s\", %s, this)%s;\n",
-			typeParameter, fun.Token, argsParamRef, lift)
+		fmt.Fprintf(w, "        {\n")
+		fmt.Fprintf(w, "            var defaultOptions = new CallOptions\n")
```

Comment:
There shouldn't be a need for these codegen changes.

The implementation of `Pulumi.Deployment.Instance.Call` should be updated to get the download URL saved on the `Resource` instance:

https://github.com/pulumi/pulumi/blob/12ec51a5f3d19971dbbf046516ab40fa290b0346/sdk/dotnet/Pulumi/Deployment/Deployment_Call.cs#L64-L70

Reply:
I think you are right that the codegen change is unnecessary. I'll revert it.